client/src/main/java/org/eclipse/hono/client/impl/HonoConnectionImpl.java

```diff
@@ -541,6 +543,10 @@ private void failConnectionAttempt(final Throwable connectionFailureCause, final
         } else if (connectionFailureCause instanceof MechanismMismatchException) {
             connectionHandler.handle(Future.failedFuture(
                     new ClientErrorException(HttpURLConnection.HTTP_UNAUTHORIZED, "no suitable SASL mechanism found for authentication with server")));
+        } else if (connectionFailureCause instanceof SSLException) {
+            connectionHandler.handle(Future.failedFuture(
+                    new ClientErrorException(HttpURLConnection.HTTP_UNAUTHORIZED,
+                            "TLS handshake with server failed: " + connectionFailureCause.getMessage())));
```

Comment:
I think it is helpful to have the root cause as well, instead of only having the top message.

Reply:
I have pushed a corresponding change ...